packages/application-shell/src/components/navbar/navbar-skeleton.tsx

```diff
@@ -0,0 +1,184 @@
+  justify-content: center;
+  align-items: center;
+  padding: ${designTokens.spacing30};
+  height: ${DIMENSIONS.header};
+  background: ${designTokens.colorAccent10};
+`;
+
+const NavBarBody = styled.div`
+  display: flex;
+  flex: 1;
+  padding: ${designTokens.spacing30};
+  flex-direction: column;
+  align-items: flex-start;
+  gap: ${designTokens.spacing40};
+  flex-shrink: 0;
+  background: #009987; // TODO: use new design token color-primary-30
+  position: relative;
+
+  // bottom gardient
+  &::after {
+    content: '';
+    position: absolute;
+    bottom: 0;
+    height: 36px;
+    width: calc(100% - 2 * ${designTokens.spacing30});
+    background: linear-gradient(180deg, rgba(0, 153, 135, 0) 0%, #009987 100%);
+  }
+`;
+
+const NavBarFooter = styled.div<TNavBarSkeletonProps>`
+  position: relative;
+  display: flex;
+  width: 100%;
+  flex-direction: column;
+  background: linear-gradient(180deg, #009987 0%, #004d44 100%);
+  padding: ${designTokens.spacing30}
+    ${(props) => (props.isExpanded ? '58px' : '0px')};
+  justify-content: center;
+  align-items: center;
+
+  // divider
+  &::before {
+    content: '';
+    position: absolute;
+    top: 0;
+    height: 1px;
+    width: calc(100% - 2 * ${designTokens.spacing30});
+    background: rgba(255, 255, 255, 0.5);
+  }
+`;
+
+const ExpandIcon = styled.div`
+  width: 40px;
+  height: 40px;
+  border-radius: ${designTokens.borderRadius4};
+  background: rgba(255, 255, 255, 0.2);
+`;
+
+const getPlacementSpecificContainerStyles = (props: TMenuItem) =>
+  props.placement === 'header'
+    ? css`
+        background: ${designTokens.colorAccent10};
+      `
+    : css`
+        padding: 12px;
+        height: 48px;
+        background: #009987; // TODO: use new design token color-primary-30
+      `;
+
+const MenuItemContainer = styled.div<TMenuItem & TNavBarSkeletonProps>`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  gap: 12px;
+  ${getPlacementSpecificContainerStyles}
+`;
+
+const MenuItemIcon = styled.div`
+  width: 24px;
+  height: 24px;
+  border-radius: ${designTokens.borderRadius4};
+  background: rgba(255, 255, 255, 0.2);
+`;
+
+const getPlacementSpecificTitleStyles = (props: TMenuItem) =>
+  props.placement === 'header'
+    ? css`
+        height: 19px;
+        width: 132px;
+      `
+    : css`
+        flex: 1;
+        height: 18px;
+      `;
+
+const MenuItemTitle = styled.div<TMenuItem>`
+  border-radius: ${designTokens.borderRadius4};
+  background: rgba(255, 255, 255, 0.2);
+  ${getPlacementSpecificTitleStyles}
+`;
+
+const MenuItemGroup = styled.div`
+  width: 100%;
+`;
+
+type TMenuItem = {
+  placement: 'header' | 'body';
+};
+
+const MenuItem = (props: TMenuItem & TNavBarSkeletonProps) => {
+  return (
+    <MenuItemContainer
+      placement={props.placement}
+      isExpanded={props.isExpanded}
+    >
+      <MenuItemIcon />
+      {props.isExpanded && <MenuItemTitle placement={props.placement} />}
+    </MenuItemContainer>
+  );
+};
```

Comment:
I'm sorry Kacper but I'm struggling to understand the `placement` attribute purpose. 

Could you please explain a little what it is used for? 🙏

Reply:
Thanks, I had hard time with this one TBH 😅
I don't think `<MenuItemContainer>` and `<MenuItemTitle>` components should be aware of the context where they're rendered.
But it also made sense to me to reuse them as part of `MenuItem` component since they're almost identical in header and body.
|  header |  body |  
|---|---|
|  <img width="280" alt="Screenshot 2023-08-11 at 15 18 00" src="https://github.com/commercetools/merchant-center-application-kit/assets/49066275/7c44c131-451f-47ae-ac5a-271cef73f48e"> |  <img width="284" alt="Screenshot 2023-08-11 at 15 18 15" src="https://github.com/commercetools/merchant-center-application-kit/assets/49066275/d93a3f4c-a635-4a19-90fb-6653ce00153e"> |

I couldn't think of any better name for the prop (`theme` doesn't really work I think) so I went for the naming borrowed from our [`<Tooltip>`](https://github.com/commercetools/ui-kit/blob/5f2886d1f29234c28c6201922d89ed7d14d5d17e/packages/components/tooltip/src/tooltip.tsx#L81-L93).

As an alternative we could have just separate styled components for both cases if that works better for you.

Reply:
I feel both visualizations are quite similar. I think the only main difference is the width of the placeholder.

I created [this PR](https://github.com/commercetools/merchant-center-application-kit/pull/3166/) to suggest what I believe is a little more simpler approach.

Let me know what you think.

Reply:
Thanks a lot for the suggestion! Merged here - [0c844b2](https://github.com/commercetools/merchant-center-application-kit/pull/3165/commits/0c844b2560208d7606f8fdd93fe177869404b6e7).